5.0 Draft/IDD/IDDs Authorization System/eu.arrowhead.authorization-management-http-json.yml

```diff
@@ -0,0 +1,518 @@
+openapi: 3.0.3
+info:
+  title: Authorization Policy Management API
+  description: |
+    This specification outlines how to realize the _Arrowhead Authorization Management_ service on top of either of the HTTP or HTTPS
+    protocols with payload encoded in JSON. The service allows for resources, data owners, service providers, or system administrators
+    to manage and express authorization rules to resources. 
+
+    To express authorization policies over resources needing protection, a _resource_ object must be created in the policy engine. This 
```

Comment:
How is a _rule_ different from a _policy_? Do we need to use both of the terms _subject_ and _consumer_? Can we stick to only _consumer_? How do _associations_ relate to the _access rights_ in the eu.arrowhead.authorization-http-json service? Are they the same?

Reply:
- A rule is something that gives or remove access rights to users, but a policy is a combination of rules that, after evaluating all of them, give a decision. This is in case one rule grants access to a certain resource, but another one explicitly denies the access, what should have preference? But I do see that the description should be more clear about it. 
- No, I should make sure that is standardised along the documents. We need, however, to choose which one to keep. My vote goes to subject because, as we discussed, we want to keep the possibility for the architect to decide to which level the authorization is checked. Naming the 'Subject' as 'Consumer' limits the idea to service exchange, which may not be the desired approach.
- Access rights are derived by Associations. So, to grant an access right to a subject, there must be an association connecting it to the desired resource that contains the desired access right. In the policy management I think they are different enough to have 2 separate terms, but I do not think associations are needed for the authorization querying, that's why they are not referred in the eu.arrowhead.authorization-http-json service.